main/tests/server/src/com/google/refine/importers/ExcelImporterTests.java

```diff
@@ -420,37 +426,33 @@ public void readMultiSheetXlsx() throws IOException {
     }
 
     @Test
-    public void testDeleteEmptyColumns() throws Exception {
-        ArrayNode sheets = ParsingUtilities.mapper.createArrayNode();
-        sheets.add(ParsingUtilities.mapper
-                .readTree("{name: \"file-source#Test Sheet 0\", fileNameAndSheetIndex: \"file-source#0\", rows: 31, selected: true}"));
+    public void testDeleteEmptyColumns() throws IOException {
+        String filename = "excel-test-file-with-empty-column.xslx";
+        List<ObjectNode> fileRecords = prepareFileRecords(xlsxFile, filename);
 
-        whenGetArrayOption("sheets", options, sheets);
-        whenGetIntegerOption("ignoreLines", options, 0);
-        whenGetIntegerOption("headerLines", options, 0);
-        whenGetIntegerOption("skipDataLines", options, 1);
-        whenGetIntegerOption("limit", options, -1);
+        ObjectNode options = ParsingUtilities.mapper.createObjectNode();
+        JSONUtilities.safePut(options, "limit", -1);
+        JSONUtilities.safePut(options, "skipDataLines", 1);
+        JSONUtilities.safePut(options, "ignoreLines", 0);
+        JSONUtilities.safePut(options, "headerLines", 0);
 
-        // This will mock the situation of deleting empty columns(col6)
-        whenGetBooleanOption("storeBlankCellsAsNulls", options, false);
-        whenGetBooleanOption("storeBlankColumns", options, false);
+        ArrayNode sheets = ParsingUtilities.mapper.createArrayNode();
+        sheets.add(ParsingUtilities.mapper.readTree(
+                String.format("{name: \"%s#Test Sheet 0\", "
+                        + "fileNameAndSheetIndex: \"%s#0\", "
+                        + "rows: 31, "
+                        + "selected: true}",
+                        filename, filename)));
+        JSONUtilities.safePut(options, "sheets", sheets);
 
-        InputStream stream = new FileInputStream(xlsxFile);
-        parseOneFile(SUT, stream);
+        JSONUtilities.safePut(options, "storeBlankCellsAsNulls", false);
+        JSONUtilities.safePut(options, "storeBlankColumns", false); // delete empty columns
+
+        parse(SUT, fileRecords, options);
 
         // We should have one less than the start due to empty column being skipped
         assertEquals(project.columnModel.columns.size(), 12);
-        // NOTE: we need to redirect through the column model because the rows will still have empty cells
-        assertEquals(project.rows.get(1).getCellValue(project.columnModel.columns.get(1).getCellIndex()), true);
-        assertEquals(project.rows.get(1).getCellValue(project.columnModel.columns.get(2).getCellIndex()), EXPECTED_DATE_TIME);
```

Comment:
Wouldn't it be worth retaining those assertions above in the new version of the test too? I agree that the `verify` ones below can be deleted though.